framework/src/utils/RankThreeTensor.C

```diff
@@ -353,3 +353,15 @@ RankTwoTensor operator*(const RealVectorValue & p, const RankThreeTensor & b)
 
   return result;
 }
+
+RealVectorValue
+RankThreeTensor::doubleContraction(const RankTwoTensor & b) const
```

Comment:
Hm, that seems bad. This is the extension of the double contraction, _not_ the dot product. It should be named accordingly. :-/